bindings/python/pyiree/compiler2/tf.py

```diff
@@ -0,0 +1,168 @@
+  OBJECT_GRAPH = "savedmodel_v2"
+  V2 = "savedmodel_v2"
+  SIGNATURE_DEF = "savedmodel_v1"
+  V1 = "savedmodel_v1"
+
+  @staticmethod
+  def parse(spec) -> "ImportType":
+    if isinstance(spec, ImportType):
+      return spec
+    spec = spec.upper()
+    if spec not in ImportType.__members__:
+      raise ValueError(f"For import_type= argument, expected one of: "
+                       f"{', '.join(ImportType.__members__.keys())}")
+    return ImportType[spec]
+
+
+class ImportOptions(CompilerOptions):
+  """Import options layer on top of the backend compiler options."""
+
+  def __init__(self,
+               exported_names: Sequence[str] = (),
+               import_only: bool = False,
+               import_type: Union[ImportType, str] = ImportType.OBJECT_GRAPH,
+               saved_model_tags: Set[str] = set(),
+               import_extra_args: Sequence[str] = (),
+               **kwargs):
+    """Initialize options from keywords.
+
+    Args:
+      exported_names: Optional sequence representing the exported names to
+        keep (object graph/v2 models only).
+      import_only: Only import the module. If True, the result will be textual
+        MLIR that can be further fed to the IREE compiler. If False (default),
+        the result will be the fully compiled IREE binary. In both cases,
+        bytes-like output is returned. Note that if the output_file= is
+        specified and import_only=True, then the MLIR form will be written to
+        the output file.
+      import_type: Type of import to perform. See ImportType enum.
+      saved_model_tags: Set of tags to export (signature def/v1 saved models
+        only).
+      import_extra_args: Extra arguments to pass to the iree-tf-import tool.
+    """
+    super().__init__(**kwargs)
+    self.exported_names = exported_names
+    self.import_only = import_only
+    self.import_type = ImportType.parse(import_type)
+    self.saved_model_tags = saved_model_tags
+    self.import_extra_args = import_extra_args
+
+
+def build_import_command_line(input_path: str,
+                              options: ImportOptions) -> List[str]:
+  """Builds a command line for invoking the import stage.
+
+  Args:
+    input_path: The input path.
+    options: Import options.
+  Returns:
+    List of strings of command line.
+  """
+  tf_import = find_tool(_TF_IMPORT_TOOL)
+  cl = [
+      tf_import,
+      input_path,
+      f"--tf-import-type={options.import_type.value}",
+      f"--tf-savedmodel-exported-names={','.join(options.exported_names)}",
+      f"--tf-savedmodel-tags={','.join(options.saved_model_tags)}",
+  ]
+  if options.import_only and options.output_file:
+    # Import stage directly outputs.
+    if options.output_file:
+      cl.append(f"-o={options.output_file}")
+  cl.extend(options.import_extra_args)
+  return cl
+
+
+def compile_saved_model(saved_model_dir: str, **kwargs):
+  """Compiles an on-disk saved model to an IREE binary."""
+  options = ImportOptions(**kwargs)
```

Comment:
I think that as things are there should be a note in `compile_saved_model` and `compile_module`'s docstrings on how `**kwargs` are routed to `ImportOptions` / `CompilerOptions`.

But it also seems like (optionally) passing an `ImportOptions` instance would surface those options to a user more transparently. (e.g. I see that `import_options` is a `pyiree.compiler.ImportOptions` (data)class from this signature, and then the `ImportOptions` init signature/doc tells me what the options are and their defaults/types). That said it _is_ more verbose if you already know the API.

Reply:
I left it as keyword args for now. Once I get this installed and see how it works with auto-completion, I feel like there are some docstring-ey things we can do to help.